### PR TITLE
REST API: Support multiple media types and MIME types when listing media

### PR DESCRIPTION
**Pull request by himanshupathak95**
Trac ticket: https://core.trac.wordpress.org/ticket/63668

This patch enhances the REST API media endpoint to allow filtering by multiple values for the `media_type` and `mime_type` parameters. This enables more flexible queries, such as requesting images and videos in a single API call.

The implementation supports both comma-separated strings and arrays for input, while maintaining full backward compatibility. Unit tests are included to cover the new functionality and edge cases.

### CODE REVIEW

File: src/wp-includes/rest-api/endpoints/class-wp-rest-attachments-controller.php [hunk 1]
```diff
@@ -82,17 +82,31 @@ protected function prepare_items_query( $prepared_args = array(), $request = nul
 			$query_args['post_status'] = 'inherit';
 		}
 
-		$media_types = $this->get_media_types();
+		$all_mime_types = array();
```
Comment (ramonjd) [inline src/wp-includes/rest-api/endpoints/class-wp-rest-attachments-controller.php:85]: Needs a `@since 6.9.0` annotation in the doc block. 🤞🏻 this makes it to 6.9!

E.g., `@since 6.9.0 Extends the  media_type and mime_type request arguments to support array values.`

File: src/wp-includes/rest-api/endpoints/class-wp-rest-attachments-controller.php [hunk 2]
```diff
@@ -82,17 +82,31 @@ protected function prepare_items_query( $prepared_args = array(), $request = nul
...
 		if ( ! empty( $request['mime_type'] ) ) {
-			$parts = explode( '/', $request['mime_type'] );
-			if ( isset( $media_types[ $parts[0] ] ) && in_array( $request['mime_type'], $media_types[ $parts[0] ], true ) ) {
-				$query_args['post_mime_type'] = $request['mime_type'];
-			}
+			$mime_type_input = is_array( $request['mime_type'] )
+				? $request['mime_type']
+				: explode( ',', $request['mime_type'] );
+
+			$all_mime_types = array_merge( $all_mime_types, array_map( 'trim', $mime_type_input ) );
```
Comment (ramonjd) [inline src/wp-includes/rest-api/endpoints/class-wp-rest-attachments-controller.php:105]: Do we still need the former check to see if the mime type prefix part is available in `$media_types`?

File: src/wp-includes/rest-api/endpoints/class-wp-rest-attachments-controller.php [hunk 3]
```diff
@@ -1285,19 +1299,24 @@ public function get_collection_params() {
...
 		$params['status']['default']       = 'inherit';
 		$params['status']['items']['enum'] = array( 'inherit', 'private', 'trash' );
-		$media_types                       = $this->get_media_types();
 
 		$params['media_type'] = array(
 			'default'     => null,
-			'description' => __( 'Limit result set to attachments of a particular media type.' ),
-			'type'        => 'string',
-			'enum'        => array_keys( $media_types ),
+			'description' => __( 'Limit result set to attachments of particular media types.' ),
```
Comment (ramonjd) [inline src/wp-includes/rest-api/endpoints/class-wp-rest-attachments-controller.php:1305]: Similar to [above](https://github.com/WordPress/wordpress-develop/pull/9211/files#r2390001814), this method's docblock needs an annotation:

**Review by ramonjd — COMMENTED** [review]
Thanks a lot for this PR!

Sorry I didn't see it sooner. 

Are you able to rebase and get it ready for a final review?

**Comment by ramonjd** [comment]
Also needs to update the API fixtures in tests/qunit/fixtures/wp-api-generated.js by running the `WP_Test_REST_Schema_Initialization` test, e.g., 

`npm run test:php -- --filter WP_Test_REST_Schema_Initialization`

That should get the tests passing.

**Comment by ramonjd** [comment]
@himanshupathak95 If you don't mind, I'd like to push this forward. I'll move your changes over to https://github.com/WordPress/wordpress-develop/pull/10054 and give you props.

**Comment by ramonjd** [comment]
Pulling across your work to https://github.com/WordPress/wordpress-develop/pull/10054 so we can get it moving.

Thank you!
